cli/cmd/plugin/unmanaged-cluster/cluster/minikube.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/config"
+
+	"k8s.io/klog/v2"
+	minikubeConfig "k8s.io/minikube/pkg/minikube/config"
+)
+
+var minikubeBinName = "minikube"
+
+type MinikubeClusterManager struct{
+	// TODO - What other minikube specific options should we implement?
+	driver string
+}
+
+// Create creates a minikube cluster with a given profile name
+func (mkcm MinikubeClusterManager) Create(c *config.UnmanagedClusterConfig) (*KubernetesCluster, error) {
+	err := mkcm.setProvierConfigs(c.ProviderConfiguration)
+	if err != nil {
+		return nil, err
+	}
+
+	profile, err := mkcm.getProfile(c.ClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("could not get minikube profiles. Error: %s", err.Error())
+	}
+
+	if profile != nil {
+		return nil, fmt.Errorf("minikube profile named: %s already exists with status: %s. This minikube profile sould be deleted before proceeding", c.ClusterName, profile.Status)
+	}
+
+	// Base start command arguments
+	args := []string{
+		"start",
+		"--driver",
+		mkcm.driver,
+		"--profile",
+		c.ClusterName,
+	}
+
+	// TODO: set node image. Do we need to host our own minikube image? Just use default gcr image?
```

Comment:
@joshrosso - what's the strategy here? The default image that `minikube` uses is one from `gcr.io/k8s-minikube/kicbase` which looks to be hosted here: https://github.com/kubernetes/minikube/tree/master/deploy/kicbase

Are we going to build and host our own image in `projects.registry.vmware.com`?

Reply:
seems like we should at some point. the question is where will it live (TCE, TF, TKG)? will we need to build it or can it just be a copy (seems like we should know what the insides of the image are)?

Reply:
We should add a minikube node image field the the TKR struct. This way it can be set and we should respect folks setting this at a config/flag level.

For this initial work, we should build our own image and point to the upstream hosted on GCR.

Lastly, can we create an issue to track building our own minikube image long-term?

Reply:
Related, our base image selection log should reflect this:

![image](https://user-images.githubusercontent.com/6200057/160891419-c9829721-8ba8-4e1e-a3bd-16c44b2117f8.png)